### PR TITLE
[Fix] fix langchain and pyspark.sql.connect.DataFrame incompatibility

### LINKED ISSUE #192: Am exploring Pyspark AI ValidationError: 1 validation error for PythonExecutor df instance of DataFrame expected (type=type_error.arbitrary_type; expected_arbitrary_type=DataFrame)
Below is the code which iam running in databricks and its throwing the below error
CODE:-
data = [('Toyota', 1849751, -9), ('Ford', 1767439, -2), ('Chevrolet', 1502389, 6),
        ('Honda', 881201, -33), ('Hyundai', 724265, -2), ('Kia', 693549, -1),
        ('Jeep', 684612, -12), ('Nissan', 682731, -25), ('Subaru', 556581, -5),
        ('Ram Trucks', 545194, -16), ('GMC', 517649, 7), ('Mercedes-Benz', 350949, 7),
        ('BMW', 332388, -1), ('Volkswagen', 301069, -20), ('Mazda', 294908, -11),
        ('Lexus', 258704, -15), ('Dodge', 190793, -12), ('Audi', 186875, -5),
        ('Cadillac', 134726, 14), ('Chrysler', 112713, -2), ('Buick', 103519, -42),
        ('Acura', 102306, -35), ('Volvo', 102038, -16), ('Mitsubishi', 102037, -16),
        ('Lincoln', 83486, -4), ('Porsche', 70065, 0), ('Genesis', 56410, 14),
        ('INFINITI', 46619, -20), ('MINI', 29504, -1), ('Alfa Romeo', 12845, -30),
        ('Maserati', 6413, -10), ('Bentley', 3975, 0), ('Lamborghini', 3134, 3),
        ('Fiat', 915, -61), ('McLaren', 840, -35), ('Rolls-Royce', 460, 7)]

auto_df = spark_ai._spark.createDataFrame(data, ["Brand", "US_Sales_2022", "Sales_Change_Percentage"])
auto_df.ai.plot()

ERROR:-

ValidationError: 1 validation error for PythonExecutor
df
  instance of DataFrame expected (type=type_error.arbitrary_type; expected_arbitrary_type=DataFrame)

### PR DESCRIPTION
**Pull request by SemyonSinchenko**
Overall idea is to introduce `DataFrameLike` object that may contain `pyspark.sql.DataFrame` or `pyspark.sql.connect.DataFrame`. It allows to workaround the problem with DBR 14.x (or any other `SparkConnect` based runtime).

Changes:
1. `DataFrameLike` dummy object
2. SparkConnect tests are separated from now
3. SparkConnect setup-script
4. SparkConnect tests are turned ON
5. Few small fixes
6. Update CI

@gengliangwang Please, review this PR. Thanks in advance!
@grundprinzip JFYI, we briefly discussed this topic today, this workaround should cover the topic while you are rising this issue in `Pydantic` / `Langchain`

Close #192
Related to #194

### CODE REVIEW

**Comment by SemyonSinchenko** [comment]
Tests are passed if one set-up "OPENAI_API_KEY": https://github.com/SemyonSinchenko/pyspark-ai/actions/runs/8070423577 (in my fork).

I will disable it again, but better to set-up this token...

File: pyspark_ai/pyspark_ai.py [hunk 1]
```diff
@@ -582,7 +582,7 @@ def plot_df(
         instruction = f"The purpose of the plot: {desc}" if desc is not None else ""
         tags = self._get_tags(cache)
         plot_chain = PythonExecutor(
-            df=df,
+            df=DataFrameLike(df),
```
Comment (grundprinzip) [inline pyspark_ai/pyspark_ai.py:585]: Interesting, was this enough for you to silence pydantic? Down the path it uses the spark sql chain and this was expecting a classic DataFrame and I had to change the annotation in the pydantic BaseModel

Comment (SemyonSinchenko) [reply pyspark_ai/pyspark_ai.py:585]: At least here it works fine. I checked with OPENAI token in my fork, tests are passed

**Comment by saikumarveera** [comment]
Hi ,

Is this issue resolved?

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks for fixing this and adding a test suite!

**Comment by gengliangwang** [comment]
I will make a new release this week.

**Comment by gengliangwang** [comment]
@SemyonSinchenko @saikumarveera FYI I just publish a new release: https://github.com/pyspark-ai/pyspark-ai/releases/tag/v0.1.21

Again, thanks for the works from @SemyonSinchenko !

**Pull request merged**